core/src/main/java/com/netflix/iceberg/BaseMetastoreTableOperations.java

```diff
@@ -129,24 +132,13 @@ protected void refreshFromMetadataLocation(String newLocation, int numRetries) {
   }
 
   @Override
-  public InputFile newInputFile(String path) {
-    return fromLocation(path, conf);
+  public String resolveMetadataPath(String fileName) {
+    return newMetadataLocation(baseLocation, fileName);
```

Comment:
I think `newMetadataLocation` is a better name than `resolveMetadataPath` because "resolve" makes me think that there is more work than just tacking on the right directory. Plus, this is always used for new files. Can you remove the `static` modifier from `newMetadataLocation` and rename `resolveMetadataPath`?

Reply:
`new` indicates the file is always going to be created, which is not necessarily the case. `resolve` I like because it doesn't necessarily indicate work to create the thing, but it indicates that it's finding the path or resolving the file name against some base path.

Reply:
> Plus, this is always used for new files.

This isn't necessarily the case in the future? How likely do we think we will only be creating files with this API?

Reply:
Fair point. How about just `metadataFileLocation` then?

Reply:
Yup that seems fine.